### DISCUSSION TITLE
split map.txt to quadrants and sectors

### DISCUSSION THREAD

**Discussion by daggs1** [endless-sky/endless-sky#7153] [Ideas]
following another PR that suggested map.txt, why not have map.txt hold 4 quadrants?
map.txt  list the 4 quadrants.
each quadrant#.txt will list the sectors within (can start from the relative start of the parent quadrant or relative to the map)
each sector#.txt will list the systems within.

thoughts?

**Comment by warp-core** [comment]
This might be useful for navigating the map file to look for stuff, but I'm not sure what the use would be for actually editing it since that's generally not done in the text file directly but instead using a map editor program

Ordering the map data in a more human navigable way creates issues here, because it might be more difficult for such programs to maintain the grouping without, for example, specifying boundaries for the quadrants and sectors, which might not take regular, easy to define shapes.

**Reply by daggs1** [reply]
I was referring to the data files rather than the game ui itself

**Comment by Zitchas** [comment]
Well, as a related sort of thing: A while  back I proposed gathering all the map related stuff into one folder in PR https://github.com/endless-sky/endless-sky/pull/5041 which was even approved by a couple of people whose opinions are respected; but was ultimately closed because the map editor can't handle subfolders or multiple data files; so we ended up merging https://github.com/endless-sky/endless-sky/pull/5163 instead, which just lumped all the Pug galaxy info into the map file, so that it would actually show up in the map editor. (Note that in #5041 this was considered to be the "least useful" solution since it just adds to the general unwieldyness of the map file, but that's the only thing the map editor can handle, so that's what we're stuck with.

If someone can update the code of the map editor to handle multiple files and subfolders, though, then this would open things up to be more human-friendly organization instead of machine-oriented usability. If this were enabled, I'd prefer to see something more along the lines of this:

1. We re-open https://github.com/endless-sky/endless-sky/pull/5041 and merge that so we get all the map data in one folder. (note that despite the age, I still have the branch, and I just updated it, so it is current)
2. We split map.txt into separate files for systems and planets. Having done this in the past for doing manual system editing, I can say that having these two separated into their own files makes editing them *vastly* easier and faster, since I don't have to keep jumping back and forth between different spots in the same file that are thousands of lines apart.
3. Once that's done, we consider splitting the map file itself into sections. I'd personally prefer to split it into regions rather than quadrants, since regions are what we actually deal with in-game; but both quadrants and regions have their own merits.

**Reply by daggs1** [reply]
where can I find the editor?

**Reply by warp-core** [reply]
https://github.com/endless-sky/endless-sky-editor
